libs/langchain-community/src/experimental/llms/chrome_ai.ts

```diff
@@ -5,25 +5,82 @@ import { GenerationChunk } from "@langchain/core/outputs";
 import { IterableReadableStream } from "@langchain/core/utils/stream";
 import { BaseLLMParams, LLM } from "@langchain/core/language_models/llms";
 
-export interface AI {
-  canCreateTextSession(): Promise<AIModelAvailability>;
-  createTextSession(options?: AITextSessionOptions): Promise<AITextSession>;
-  defaultTextSessionOptions(): Promise<AITextSessionOptions>;
+export interface AILanguageModelFactory {
+  create(options?: AILanguageModelCreateOptions): Promise<AILanguageModel>;
+  capabilities(): Promise<AILanguageModelCapabilities>;
 }
 
-export interface AITextSession {
-  prompt(input: string): Promise<string>;
-  promptStreaming(input: string): ReadableStream;
+export interface AILanguageModel extends EventTarget {
+  prompt(input: AILanguageModelPromptInput, options?: AILanguageModelPromptOptions): Promise<String>;
+  promptStreaming(input: AILanguageModelPromptInput, options?: AILanguageModelPromptOptions): ReadableStream;
+
+  countPromptTokens(input: AILanguageModelPromptInput, options?: AILanguageModelPromptOptions): Promise<number>;
```

Comment:
I have updated the interface to the latest definition, we may implement these if it make sense.

Reply:
Given this will likely change again, is it worth adding all of this if we're not using it?

Reply:
Eh it's experimental I suppose so it's fine